env.go

```diff
@@ -128,7 +129,10 @@ func parseVars(v reflect.Value) []Var {
 			if ok {
 				prefix = value
 			}
-			for _, v := range parseVars(field) {
+			if opts != nil {
+				prefix += opts.NameSep
```

Comment:
We should append the prefix only if the `env` tag is present (line 130). Given the following code:

```go
var cfg struct {
	DB struct {
		Host string `env:"HOST"`
		Port int    `env:"PORT"`
	} `env:"DB"`
	HTTP struct {
		Port string `env:"HTTP_PORT"`
	}
}
if err := env.Load(&cfg, &env.Options{NameSep: "_"}); err != nil {
	// ...
}
```

Would lookup for `_HTTP_PORT` instead of `HTTP_PORT`.

Also, `opts` won't be nil since we initialize it at the start of `Load`.